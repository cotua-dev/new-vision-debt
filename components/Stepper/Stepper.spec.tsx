import { cleanup, render, RenderResult } from '@testing-library/react';
import { Stepper } from './index';

describe('Stepper Component', (): void => {
    let renderResult: RenderResult;

    beforeEach((): RenderResult => renderResult = render(<Stepper/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    afterEach(cleanup);
});
