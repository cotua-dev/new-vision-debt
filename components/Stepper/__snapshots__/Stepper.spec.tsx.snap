// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Component renders 1`] = `
<section
  class="debt-advisors-stepper"
>
  <small>
    1 of 5
  </small>
  <div
    class="steps-wrapper"
  >
    <div
      class="step "
    >
      <strong
        class="question"
      >
        What are you looking to do?
      </strong>
      <div
        class="multiple-choice-wrapper"
      >
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Reduce your monthly payments
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Consolidate your payments
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Reduce your interest rates
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            All of the above
          </strong>
        </button>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What kind of debts do you have?
      </strong>
      <div
        class="multiple-choice-wrapper"
      >
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Credit cards
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Personal loans
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Medical bills
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            All of the above
          </strong>
        </button>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        Are you behind on your payments?
      </strong>
      <div
        class="multiple-choice-wrapper"
      >
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Yes - more than 60 days
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            Yes - more than 30 days
          </strong>
        </button>
        <button
          class="choice-button "
          type="button"
        >
          <strong
            class="choice-text"
          >
            No
          </strong>
        </button>
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What is your monthly income?
      </strong>
      <div
        class="currency-field-wrapper"
      >
        <input
          class="currency-field"
          inputmode="numeric"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="step hidden"
    >
      <strong
        class="question"
      >
        What is the total amount of your unsecured debt?
      </strong>
      <div
        class="currency-field-wrapper"
      >
        <input
          class="currency-field"
          inputmode="numeric"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="stepper-controls-wrapper"
  >
    <button
      class="previous-button"
      disabled=""
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-left caret"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 448c-8.188 0-16.38-3.125-22.62-9.375l-160-160c-12.5-12.5-12.5-32.75 0-45.25l160-160c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L77.25 256l137.4 137.4c12.5 12.5 12.5 32.75 0 45.25C208.4 444.9 200.2 448 192 448z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="next-button"
      disabled=""
      type="button"
    >
      Next
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right caret"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 448c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L178.8 256L41.38 118.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l160 160c12.5 12.5 12.5 32.75 0 45.25l-160 160C80.38 444.9 72.19 448 64 448z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</section>
`;
